### PR TITLE
use TaskWorkerConfig to enable use of V2 bindings

### LINKED ISSUE #8571: introduce a software switch for using HTC bindings v2
- [x] a switch in TaskWorkerConfig.py
- [x] pass this info to Scheduler via a file in the spool directory. Each python script will check that file presence at the beginning before doing `import htcondor` or `import htcondor2 as htcondor`
  * Changed to use an env.variable defined in the submit JDL of the job submitted by DagmanSubmitter

### PR DESCRIPTION
**Pull request by belforte**
fix #8571 
#### Summary:
* switch v1/v2 works
* Running with v1 is OK.
* Testing highlighted another problem in using v2, will fix that separately.

### CODE REVIEW

**Comment by belforte** [comment]
validation with v1 bindings passed
- [ ] https://github.com/dmwm/CRABServer/issues/8599

File: scripts/get_pool_status.py [hunk 1]
```diff
@@ -3,18 +3,19 @@
 from __future__ import print_function
 import re
 import pprint
-import optparse
+import optparse  # pylint: disable=deprecated-module
 
-import classad
-import htcondor
+import htcondor2 as htcondor
```
Comment (novicecpp) [inline scripts/get_pool_status.py:8]: Did you also need `if 'useHtcV2' in os.environ:` for this file?

File: scripts/task_process/cache_status.py [hunk 2]
```diff
@@ -68,8 +73,8 @@ def insertCpu(event, info):
             info['TotalUserCpuTimeHistory'][-1] = float(event['RemoteUserCpu'])
 
 
-nodeNameRe = re.compile("DAG Node: Job(\d+(?:-\d+)?)")
-nodeName2Re = re.compile("Job(\d+(?:-\d+)?)")
+nodeNameRe = re.compile(r"DAG Node: Job(\d+(?:-\d+)?)")
+nodeName2Re = re.compile(r"Job(\d+(?:-\d+)?)")
```
Comment (novicecpp) [inline scripts/task_process/cache_status.py:78]: Are these still the same?

File: src/python/ServerUtilities.py [hunk 3]
```diff
@@ -198,7 +198,7 @@
                           ]}
 
 
-def NEW_USER_SANDBOX_EXCLUSIONS(tarmembers):  # TODO move to CRABClient !!!
+def NEW_USER_SANDBOX_EXCLUSIONS(tarmembers):  # This is only used in CRABClient !!! MOVE IT THERE
```
Comment (novicecpp) [inline src/python/ServerUtilities.py:201]: Suggestion: Please refrain from changing code/comments that are unrelated to the PR.

File: src/python/TaskWorker/Main.py [hunk 4]
```diff
@@ -1,23 +1,21 @@
...
+Top driver for TaskWorker service
+"""
+from optparse import OptionParser  # pylint: disable=deprecated-module
 import signal
 import os
 import logging
 import sys
 
 from WMCore.Configuration import loadConfigurationFile
 from TaskWorker.WorkerExceptions import ConfigException
```
Comment (novicecpp) [inline src/python/TaskWorker/Main.py:11]: Shall we also move this line to late binding? Just in case.
And maybe put a comment here that you need to do late import for import crab relate code.

**Comment by belforte** [comment]
WOW. What a careful review. Thanks Wa. I will address the minor comments.
As to testing v2 version, I did and it works at first sight, but I have found one more issue to fix https://github.com/dmwm/CRABServer/issues/8597
I'd rather like to close the "a switch for v2" chapter and do more changes and testing on the master branch.

File: src/python/TaskWorker/Main.py (hunk 4 above)
Comment (belforte) [reply src/python/TaskWorker/Main.py:11]: My apologies. I do not understand which line is `this line` and where you suggest to move it to. OK for the comment.

Comment (novicecpp) [reply src/python/TaskWorker/Main.py:11]: Ah...I mean line 11. 
```python
 from TaskWorker.WorkerExceptions import ConfigException
```
I thought when I click at line 11 (which does not contain any diff) it will only show line 11.

Comment (belforte) [reply src/python/TaskWorker/Main.py:11]: got it. Yes. Thanks

File: scripts/task_process/cache_status.py (hunk 2 above)
Comment (belforte) [reply scripts/task_process/cache_status.py:78]: I simply added `r` (means `raw`) before the string to please pylint. IIUC it tells python not to inteprete the `\` as an escape. Which it would not do in this case, but it still looks better.

File: src/python/ServerUtilities.py (hunk 3 above)
Comment (belforte) [reply src/python/ServerUtilities.py:201]: I know.. but I wanted go have a clean pylint report

File: scripts/get_pool_status.py (hunk 1 above)
Comment (belforte) [reply scripts/get_pool_status.py:8]: we do not use this file anywhere. I am keeping it around as example just in case we need something like this in the future. Either way it will need testing and validation in case of any future use.

File: scripts/dag_bootstrap_startup.sh [hunk 5]
```diff
@@ -33,6 +33,11 @@ fi
 srcname=$0
 env > ${srcname%.sh}.env
 
+if [ "X$useHtcV2" != "X" ]; then
```
[the remark this answers is not in the document]
Comment (belforte) [reply scripts/dag_bootstrap_startup.sh:36]: done

File: scripts/dag_bootstrap_startup.sh [hunk 6]
```diff
@@ -240,8 +245,13 @@ Error         = task_process/daemon.err.\$(Cluster).\$(Process)
...
 Queue 1
 EOF
-        # TODO - remove chmod
-        chmod 777 task_process/task_proc_wrapper.sh
+
+if [ "X$useHtcV2" != "X" ]; then
+  cat >> task_process/daemon.jdl << EOF
+environment = "useHtcV2=True"
+EOF
+fi
```
[the remark this answers is not in the document]
Comment (belforte) [reply scripts/dag_bootstrap_startup.sh:253]: thanks for notificing the missing indent.
I would have thought the code to be self-explanatory: if useHTCV2 is set in the environment then add a line in the submission JDL of the task_process so that it is set also in the environment of that job :man_shrugging: 

Anyhow I am adding that as a comment as well.

File: src/python/TaskWorker/Main.py (hunk 4 above)
Comment (belforte) [reply src/python/TaskWorker/Main.py:11]: Done

**Review by novicecpp — APPROVED** [review]

**Comment by belforte** [comment]
one last round of validation after the changes..
### pipeline (here it uses v1)
- [ ] https://gitlab.cern.ch/crab3/CRABServer/-/pipelines/7908895 

### Jenkins
#### v1
- [ ] CCV https://github.com/dmwm/CRABServer/issues/8600#issuecomment-2278163089
- [ ] ST https://github.com/dmwm/CRABServer/issues/8601

#### v2
- [ ] CCV
- [ ] ST

**Comment by belforte** [comment]
OK. Let's go
